### PR TITLE
nfd-master: use separate k8s api clients for each updater

### PR DESCRIPTION
**Pull request by marquiz**
Sharing the same client between updater threads virtually serializes access, in practice making the effective parallelism close to 1.

With this patch, in my bench cluster of 300 nodes, the time taken by updating all nodes drops from ~2 minutes to ~12 seconds (with the default parallelism of 10 node updater threads). This demonstrates the 10-fold increased parallelism from ~1 to 10.

There might be other solutions that could be explored, e.g. caching nodes with an indexer/lister but otoh nfd doesn't necessarily need/want to watch every little change in each node. We only need to get the node when something in our own CRDs change (we don't react to any changes in the node object itself). Using multiple clients was the most obvious choice to solve the problem for now.

### CODE REVIEW

**Comment by marquiz** [comment]
Depends on #1652 
/hold

/assign @ArangoGutierrez @TessaIO

**Comment by marquiz** [comment]
Depends on #1655

**Comment by marquiz** [comment]
#1655 merged
/unhold

Should not need rebase 🤨

File: pkg/nfd-master/node-updater-pool.go
Comment (ArangoGutierrez) [inline pkg/nfd-master/node-updater-pool.go:77]: I am thinking, should we do the if first and defer to an `else` in case `u.nfdMaster.kubeconfig == nil`

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]
Just one nit/question

File: pkg/nfd-master/node-updater-pool.go
Comment (marquiz) [reply pkg/nfd-master/node-updater-pool.go:77]: This was a short version to initialize the `cli` variable. An alternative would be like:

```golang
	var cli k8sclient.Interface
	if u.nfdMaster.kubeconfig != nil {
		// For normal execution, initialize a separate api client for each updater
		cli = k8sclient.NewForConfigOrDie(u.nfdMaster.kubeconfig)
	} else {
		// For tests, re-use the api client from nfd-master
		cli := u.nfdMaster.k8sClient
	}
```

I'm ok with either

Comment (ArangoGutierrez) [reply pkg/nfd-master/node-updater-pool.go:77]: I like it, not a blocker

**Review by ArangoGutierrez — APPROVED** [review]

File: pkg/nfd-master/node-updater-pool.go
Comment (marquiz) [reply pkg/nfd-master/node-updater-pool.go:76]: I changed the code to what I suggested

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

**Pull request merged**
